.github/workflows/schedule.py

```diff
@@ -75,7 +89,105 @@ def _load_issue_template(path: str) -> tuple[dict[str, str], str]:
+    configured shcedule for the provided template. This logs the schedule check for
+    each, and it only returns what falls within the schedule window.
+    """
+    log.info('Checking configured schedule time window for %r', template)
+    if 0 == (now - start).days % period_days and now.hour == at_hour:
+        log.info('Scheduling %r, it falls within schedule window', template)
+        return template, now
+    else:
+        log.info('Current time is outside of the configured schedule window')
+
+
 if __name__ == '__main__':
     logging.basicConfig(level=logging.INFO,
                         format='%(asctime)s %(levelname)+7s %(name)s: %(message)s')
-    create_upgrade_issue()
+
+    args = sys.argv[1:]
+    parser = argparse.ArgumentParser(
+        description='Dryrun to display the command used by GH Actions to create an issue'
+    )
+    parser.add_argument('--dry-run',
+                        metavar='MOCKED_DATE',
+                        type=lambda s: datetime.strptime(s, '%Y-%m-%dT%H:%M'),
+                        help='Dry run based on mock date to verify the template issue configuration')
+    args = parser.parse_args(args)
+
+    now = datetime.now(tz)
+    if args.dry_run:
+        dt = args.dry_run
+        now = now.replace(year=dt.year,
+                          month=dt.month,
+                          day=dt.day,
+                          hour=dt.hour,
+                          minute=dt.minute)
+
+    templates = [
+        # Pick any date for `start` and an issue will be created on that date,
+        # and thereafter based on the given period, but only in the future.
+        # Note that this doesn't mean that the start date has to lie in the
+        # future. As an example, …
+        _create(template, start, period, hour) for template, start, period, hour in [
+            (
+                'upgrade.md',  # … the schedule for upgrade.md is set …
+                datetime(2023, 11, 27, tzinfo=tz),  # … for Monday …
+                one_week * 2,  # … every other week (14 days) …
+                9  # … at 9am.
+            ),
+            (
+                'promotion.md',
+                datetime(2024, 2, 27, tzinfo=tz),
+                one_week,
+                9
+            ),
+            (
+                'gitlab_backups.md',
+                datetime(2024, 2, 26, tzinfo=tz),
+                one_week * 2,
+                9
+            ),
+            (
+                'opensearch_updates.md',
+                datetime(2024, 2, 26, tzinfo=tz),
+                one_week * 2,
+                9
+            ),
+            (
+                'fedramp_inventory_review.md',
+                datetime(2024, 3, 1, tzinfo=tz),
+                one_week * 4,
+                9
+            ),
+            (
+                'audited_events_rule_set_review.md',
+                datetime(2024, 4, 3, tzinfo=tz),
+                364,  # Every year, (364 days)
```

Comment:
This does not work. A calendar year is usually 365 days, 366 days for leap years.